pynats/client.py

```diff
@@ -279,7 +280,11 @@ def _readline(self, *, size: int = None) -> bytes:
         read = io.BytesIO()
 
         while True:
-            line = cast(bytes, self._socket_file.readline())
+            raw_bytes = self._socket_file.readline()
+            if not raw_bytes:
+                raise RuntimeError("unable to read from socket")
```

Comment:
Exception is much better. But please introduce a new exception [here](https://github.com/Gr1N/nats-python/blob/master/pynats/exceptions.py) for your case and use it.

Also, it would be great to cover that case with tests.

Reply:
Updated as requested, thanks!